src/app/app.component.ts

```diff
@@ -26,7 +29,10 @@ export class AppComponent {
    *
    * @param data The data state.
    */
-  constructor(readonly data: DataState) { data.listData$.subscribe(console.log); data.aggregateData$.subscribe(console.log); }
+  constructor(readonly data: DataState, readonly dataSourceService: DataSourceService) {
```

Comment:
Don't import DataSourceService instead add a method on the DataState for fetching the image viewer data.